File: extract_pdf.js
const fs = require('fs');
const pdf = require('pdf-parse');

const pdfPath = 'C:\\Users\\oren weiss\\Downloads\\ukEv13158223.pdf';

let dataBuffer = fs.readFileSync(pdfPath);

pdf(dataBuffer).then(function (data) {
    fs.writeFileSync('haggadah_extracted.txt', data.text);
    console.log('Text extracted to haggadah_extracted.txt');
}).catch(err => {
    console.error('Error parsing PDF:', err);
});

File: extract_pdf.py
import pypdf
import json

pdf_path = r'C:\Users\oren weiss\Downloads\ukEv13158223.pdf'

reader = pypdf.PdfReader(pdf_path)
full_text = ""
for page in reader.pages:
    full_text += page.extract_text() + "\n"

with open('haggadah_full_raw.txt', 'w', encoding='utf-8') as f:
    f.write(full_text)

print("Extraction complete. Raw text saved to haggadah_full_raw.txt")
